### DISCUSSION TITLE
Request to create energy for pv1 and pv2

### DISCUSSION THREAD

**Discussion by Anxiumer** [wills106/homeassistant-solax-modbus#125] [Ideas]
I've got a split east/west setup and had split my forecast based on the panels in different places, could we add in energy for both pv1 and pv2 so that we can match up those on their own to the respective forecast? I think previously I'd used an "Integration" sensor to generate the guestimate - is this something that could be added?

If I wanted to contribute do I just create a branch and pullrequest?

**Comment by wills106** [comment]
Feel free to submit a PR. Otherwise I can make the change in the next day or two.

**Reply by Anxiumer** [reply]
I won't be that quick - though adding a helper of type "Riemann sum integral" was just super easy so maybe that is the right idea anyway as I'm not convinced there's an easy way to do it in to component. Would you calculate each refresh the additional energy used by pv1/pv2?

**Reply by wills106** [reply]
I can't guarantee PV1 & PV2 power will exactly match "SolaX Today's Solar Energy" But it should be close enough for what you require.

**Reply by blb4github** [reply]
Not sure if this is the best place, but I have doubt about the "Solax Today's Solar Energy" value. I have batteries connected to my inverter and to me it looks like this sensor value is the AC Energy output of the inverter, not the Solar Energy. See attached graphs:
![Solax](https://github.com/wills106/homeassistant-solax-modbus/assets/35904565/38bfd45b-7000-403e-8c1f-ede4736b3588)

As far as I can see the same counts for "SolaX Total Solar Energy". Would it be better to make an issue?
